hazelcast/src/main/java/com/hazelcast/internal/affinity/ThreadAffinityLoader.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.affinity;
+
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.spi.properties.GroupProperty;
+
+import static com.hazelcast.internal.affinity.ThreadAffinity.Group.IO;
+import static com.hazelcast.internal.affinity.ThreadAffinity.Group.PARTITION_THREAD;
+import static com.hazelcast.internal.affinity.ThreadAffinityParamsHelper.isAffinityEnabled;
+import static com.hazelcast.util.EmptyStatement.ignore;
+import static com.hazelcast.util.OsHelper.OS;
+import static com.hazelcast.util.OsHelper.isUnixFamily;
+import static java.util.Collections.disjoint;
+
+@SuppressWarnings({"checkstyle:npathcomplexity", "checkstyle:cyclomaticcomplexity"})
+public final class ThreadAffinityLoader {
```

Comment:
The name is confusing because it doesn't load a ThreadAffinity.
It loads an AffinitySupportExtension. 

Imho this logic should be merged into the AffinitySupportExtension class.